test/servinge2e/servicemesh_test.go

```diff
@@ -594,3 +603,262 @@ func TestKsvcWithServiceMeshJWTDefaultPolicy(t *testing.T) {
+	certPool := x509.NewCertPool()
+	certPool.AppendCertsFromPEM(exampleSecret.Data["tls.crt"])
+
+	// Deploy a cluster-local ksvc "hello"
+	ksvc := test.Service("hello", serviceMeshTestNamespaceName, "openshift/hello-openshift", nil)
+	ksvc.ObjectMeta.Labels = map[string]string{
+		serving.VisibilityLabelKey: serving.VisibilityClusterLocal,
+	}
+	ksvc = withServiceReadyOrFail(caCtx, ksvc)
+
+	// Create the Istio Gateway for traffic via istio-ingressgateway
+	// The secret and its mounts are specified in the example SMCP in hack/lib/mesh.bash
+	//
+	//       istio-ingressgateway:
+	//        secretVolumes:
+	//        - mountPath: /custom.example.com
+	//          name: custom-example-com
+	//          secretName: custom.example.com
+	//
+	defaultGateway := test.IstioGatewayWithTls("default-gateway",
+		serviceMeshTestNamespaceName,
+		customDomain,
+		"/custom.example.com/tls.key",
+		"/custom.example.com/tls.crt",
+	)
+	defaultGateway = test.CreateIstioGateway(caCtx, defaultGateway)
+
+	// Create the Istio VirtualService to rewrite the host header of a custom domain with the ksvc's svc hostname
+	virtualService := test.IstioVirtualServiceForKnativeServiceWithCustomDomain(ksvc, defaultGateway.GetName(), customDomain)
+	virtualService = test.CreateIstioVirtualService(caCtx, virtualService)
+
+	// Create the Istio ServiceEntry for ksvc's svc hostname routing towards the knative kourier-internal gateway
+	serviceEntry := test.IstioServiceEntryForKnativeServiceTowardsKourier(ksvc)
+	serviceEntry = test.CreateIstioServiceEntry(caCtx, serviceEntry)
+
+	// Create the OpenShift Route for the custom domain pointing to the istio-ingressgateway
+	// Note, this one is created in the service mesh namespace ("istio-system"), not the test namespace
+	route := &routev1.Route{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      "hello",
+			Namespace: serviceMeshNamespace,
+		},
+		Spec: routev1.RouteSpec{
+			Host: customDomain,
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromInt(8443),
+			},
+			TLS: &routev1.TLSConfig{
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyNone,
+				Termination:                   routev1.TLSTerminationPassthrough,
+			},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: "istio-ingressgateway",
+			},
+		},
+	}
+	route, err = caCtx.Clients.Route.Routes(serviceMeshNamespace).Create(route)
+	if err != nil {
+		t.Fatalf("Error creating OpenShift Route: %v", err)
+	}
+
+	caCtx.AddToCleanup(func() error {
+		t.Logf("Cleaning up OpenShift Route %s", route.GetName())
+		return caCtx.Clients.Route.Routes(route.Namespace).Delete(route.Name, &meta.DeleteOptions{})
+	})
+
+	// Do a spoofed HTTP request.
+	// Note, here we go via the OpenShift Router IP address, not kourier as usual with the "spoof" client.
+	ips := lookupOpenShiftRouterIP(caCtx)
+	ip := ips[0].String()
+	t.Logf("Using the following IP address as the OpenShift Router address: %s", ips[0].String())
+
```

Comment:
I wonder that we want to verify that `HTTP` access is not available as we set `InsecureEdgeTerminationPolicyNone` to the Route.
How do you think @maschmid ?

Reply:
sure, we can do that...